cmd/platform/sampledata.go

```diff
@@ -0,0 +1,628 @@
+	if rand.Intn(5) == 0 {
+		nickname = fake.Company()
+	}
+
+	// Half of users skip tutorial
+	tutorialStep := "999"
+	switch rand.Intn(6) {
+	case 1:
+		tutorialStep = "1"
+	case 2:
+		tutorialStep = "2"
+	case 3:
+		tutorialStep = "3"
+	}
+
+	teams := []app.UserTeamImportData{}
+	possibleTeams := []string{}
+	for teamName := range teamsAndChannels {
+		possibleTeams = append(possibleTeams, teamName)
+	}
+	sort.Strings(possibleTeams)
+	for x := 0; x < teamMemberships; x++ {
+		if len(possibleTeams) == 0 {
+			break
+		}
+		position := rand.Intn(len(possibleTeams))
+		team := possibleTeams[position]
+		possibleTeams = append(possibleTeams[:position], possibleTeams[position+1:]...)
+		if teamChannels, err := teamsAndChannels[team]; err == true {
+			teams = append(teams, createTeamMembership(channelMemberships, teamChannels, &team))
+		}
+	}
+
+	user := app.UserImportData{
+		ProfileImage:       profileImage,
+		Username:           &username,
+		Email:              &email,
+		Password:           &password,
+		Nickname:           &nickname,
+		FirstName:          &firstName,
+		LastName:           &lastName,
+		Position:           &position,
+		Roles:              &roles,
+		Teams:              &teams,
+		UseMilitaryTime:    &useMilitaryTime,
+		CollapsePreviews:   &collapsePreviews,
+		MessageDisplay:     &messageDisplay,
+		ChannelDisplayMode: &channelDisplayMode,
+		TutorialStep:       &tutorialStep,
+	}
+	return app.LineImportData{
+		Type: "user",
+		User: &user,
+	}
+}
+
+func createTeamMembership(numOfchannels int, teamChannels []string, teamName *string) app.UserTeamImportData {
+	roles := "team_user"
+	if rand.Intn(5) == 0 {
+		roles = "team_user team_admin"
+	}
+	channels := []app.UserChannelImportData{}
+	teamChannelsCopy := []string{}
+	for _, value := range teamChannels {
+		teamChannelsCopy = append(teamChannelsCopy, value)
+	}
+	for x := 0; x < numOfchannels; x++ {
+		if len(teamChannelsCopy) == 0 {
+			break
+		}
+		position := rand.Intn(len(teamChannelsCopy))
+		channelName := teamChannelsCopy[position]
+		teamChannelsCopy = append(teamChannelsCopy[:position], teamChannelsCopy[position+1:]...)
+		channels = append(channels, createChannelMembership(channelName))
+	}
+
+	return app.UserTeamImportData{
+		Name:     teamName,
+		Roles:    &roles,
+		Channels: &channels,
+	}
+}
+
+func createChannelMembership(channelName string) app.UserChannelImportData {
+	roles := "channel_user"
+	if rand.Intn(5) == 0 {
+		roles = "channel_user channel_admin"
+	}
+	favorite := rand.Intn(5) == 0
+
+	return app.UserChannelImportData{
+		Name:     &channelName,
+		Roles:    &roles,
+		Favorite: &favorite,
+	}
+}
+
+func createTeam(idx int) app.LineImportData {
+	displayName := fake.Word()
+	name := fmt.Sprintf("%s-%d", fake.Word(), idx)
+	allowOpenInvite := rand.Intn(2) == 0
+
+	description := fake.Paragraph()
+	if len(description) > 255 {
+		description = description[0:255]
+	}
+
+	teamType := "O"
+	if rand.Intn(2) == 0 {
+		teamType = "I"
```

Comment:
Need to check, but I think team types other than "O" are legacy and not used any more.